state-chain/runtime/src/lib.rs

```diff
@@ -1285,116 +1287,109 @@ type PalletMigrations = (
-	channel_id: u64,
-	asset: <T::TargetChain as cf_chains::Chain>::ChainAsset,
-) -> Result<(), DispatchError> {
-	let channel =
-		cf_chains::DepositChannel::<T::TargetChain>::generate_new::<T::AddressDerivation>(
-			channel_id, asset,
-		)
-		.map_err(|_| DispatchError::Other("Failed to generate new deposit channel"))?;
-	pallet_cf_ingress_egress::DepositChannelPool::<T, I>::insert(
-		channel.channel_id,
-		channel.clone(),
-	);
-
-	let (channel_id, address, ..) =
-		pallet_cf_ingress_egress::Pallet::<T, I>::request_liquidity_deposit_address(
-			AccountId32::new(CHANNEL_ACCOUNT),
-			asset,
-			Default::default(),
-			cf_chains::ForeignChainAddress::Sol(SolAddress(bs58_array(
-				"Hqn5iCLXYm2pP2iqzVAsvsNAFw1kuvEGXxG3JAqwfsEq",
-			))),
-		)?;
-
-	frame_support::ensure!(channel_id == channel.channel_id, "Channel IDs don't match");
-	frame_support::ensure!(
-		<<T::TargetChain as cf_chains::Chain>::ChainAccount>::try_from(address)
-			.map_err(|_| "Failed to convert address")? ==
-			channel.address,
-		"Addresses don't match"
-	);
-
-	Ok(())
+use cf_primitives::chains::assets::btc::Asset as BtcAsset;
+
+pub struct BoostRecovery;
+
+const RECOVERY_TX_ID: H256 =
+	H256(hex_literal::hex!("a250b52718e91a1d7184690d0985f068db21d4a949254a8a994e6c412bc66612"));
+const SALT: u32 = 29875;
+const PUBKEY: [u8; 32] =
+	hex_literal::hex!("af7ee887264c85f3eb84468c7cae0bc6b40d23732f94d5db591226654dcc7803");
+const POOL_TIER: u16 = 5;
+
+impl BoostRecovery {
+	/// Only run this once, and only on mainnet. Use the spec_version and utxo id as a markers.
+	fn should_run() -> bool {
+		use cf_runtime_upgrade_utilities::genesis_hashes::{genesis_hash, BERGHAIN};
+		genesis_hash::<Runtime>() == BERGHAIN &&
+			VERSION.spec_version == 179 &&
+			pallet_cf_environment::BitcoinAvailableUtxos::<Runtime>::get()
+				.iter()
+				.all(|utxo| utxo.id.tx_id != RECOVERY_TX_ID)
+	}
 }
 
-impl frame_support::traits::OnRuntimeUpgrade for IngressMigration {
+impl frame_support::traits::OnRuntimeUpgrade for BoostRecovery {
 	fn on_runtime_upgrade() -> Weight {
-		if should_run() {
-			let previous_safe_mode =
-				pallet_cf_environment::RuntimeSafeMode::<Runtime>::mutate(|safe_mode| {
-					core::mem::replace(&mut safe_mode.ingress_egress_solana.deposits_enabled, true)
-				});
-			let _ = reopen::<Runtime, SolanaInstance>(CHANNEL_ID, SolAsset::Sol).map_err(|e| {
-				log::warn!("⛔️ Failed to reopen channel: {:?}", e);
-			});
-			let _ = pallet_cf_environment::RuntimeSafeMode::<Runtime>::mutate(|safe_mode| {
-				core::mem::replace(
-					&mut safe_mode.ingress_egress_solana.deposits_enabled,
-					previous_safe_mode,
-				)
-			});
+		if !Self::should_run() {
+			return Weight::zero();
 		}
+
+		pallet_cf_ingress_egress::Pallet::<Runtime, BitcoinInstance>::recover_boost_deposit(
+			BtcAsset::Btc,
+			POOL_TIER,
+			migrations::boost_recovery::PREWITNESSED_DEPOSIT_ID,
+			migrations::boost_recovery::scaled_boost_contributions_from_raw(),
+			Default::default(),
```

Comment:
Does this mean we had no withdrawals at the time the deposit was considered "lost"?

Reply:
Yes, it was empty.

Reply:
But I suppose it would have been more correct to take it from the snapshot.